Unable to create 3.10.1 cluster

I've pulled a clean clone of aws-eda-slurm-cluster.
After a source setup.sh, I'm able to create a 3.11.1 cluster with the install.sh --cdk-cmd create 
I then create a new config yml file, only changing the cluster version and my custom ami to be a 3.10.1 ami.
Here is the diff of the config files:

```
diff tsi3fcs-x86_instance_types.yml tsi3-3-10-1-x86_instance_types.yml
11c11
< StackName: tsi3fcs-config
---
> StackName: tsi3-3-10-1-config
42c42
<     Version: 3.11.1
---
>     Version: 3.10.1
46c46
<       CustomAmi: ami-0d68c6538cd916c25  # pcluster-3-11-1-Rocky-8-x86-64-ami-0d0023cdec9d16c99 2024-10-24T03-12-55.412Z
---
>       CustomAmi: ami-03ff401c2420be54f   # pcluster-3-10-1-Rocky-8-10-x86-64-ami-0dae235de369c403c
```

I am not able to create a 3.10.1 cluster. Error message in CloudWatch logs, [/aws/lambda/tsi3-3-10-1-config-CreateParallelClusterConfig](https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#logsV2:log-groups/log-group/$252Faws$252Flambda$252Ftsi3-3-10-1-config-CreateParallelClusterConfig) complains about unable to import module rpds.rpds:

-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
|   timestamp   |                                                                                                message                                                                                                |
|---------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 1730609451371 | INIT_START Runtime Version: python:3.12.v36 Runtime Version ARN: arn:aws:lambda:us-west-2::runtime:188d9ca2e2714ff5637bd2bbe06ceb81ec3bc408a0f277dab104c14cd814b081                                   |
| 1730609452483 | [ERROR] Runtime.ImportModuleError: Unable to import module 'CreateParallelClusterConfig': No module named 'rpds.rpds' Traceback (most recent call last):                                              |
| 1730609452564 | INIT_REPORT Init Duration: 1193.71 ms Phase: init Status: error Error Type: Runtime.ImportModuleError                                                                                                 |
| 1730609453572 | [ERROR] Runtime.ImportModuleError: Unable to import module 'CreateParallelClusterConfig': No module named 'rpds.rpds' Traceback (most recent call last):                                              |
| 1730609453648 | INIT_REPORT Init Duration: 1069.94 ms Phase: invoke Status: error Error Type: Runtime.ImportModuleError                                                                                               |
| 1730609453648 | START RequestId: fa33bc2e-d8d1-4f6b-b369-81dc3079d710 Version: $LATEST                                                                                                                                |
| 1730609453663 | END RequestId: fa33bc2e-d8d1-4f6b-b369-81dc3079d710                                                                                                                                                   |
| 1730609453663 | REPORT RequestId: fa33bc2e-d8d1-4f6b-b369-81dc3079d710 Duration: 1083.74 ms Billed Duration: 1084 ms Memory Size: 2048 MB Max Memory Used: 80 MB Status: error Error Type: Runtime.ImportModuleError  |
| 1730609520786 | [ERROR] Runtime.ImportModuleError: Unable to import module 'CreateParallelClusterConfig': No module named 'rpds.rpds' Traceback (most recent call last):                                              |
| 1730609520874 | INIT_REPORT Init Duration: 1070.40 ms Phase: invoke Status: error Error Type: Runtime.ImportModuleError                                                                                               |
| 1730609520874 | START RequestId: fa33bc2e-d8d1-4f6b-b369-81dc3079d710 Version: $LATEST                                                                                                                                |
| 1730609520885 | END RequestId: fa33bc2e-d8d1-4f6b-b369-81dc3079d710                                                                                                                                                   |
| 1730609520885 | REPORT RequestId: fa33bc2e-d8d1-4f6b-b369-81dc3079d710 Duration: 1081.01 ms Billed Duration: 1082 ms Memory Size: 2048 MB Max Memory Used: 80 MB Status: error Error Type: Runtime.ImportModuleError  |
| 1730609636990 | [ERROR] Runtime.ImportModuleError: Unable to import module 'CreateParallelClusterConfig': No module named 'rpds.rpds' Traceback (most recent call last):                                              |
| 1730609637064 | INIT_REPORT Init Duration: 1036.39 ms Phase: invoke Status: error Error Type: Runtime.ImportModuleError                                                                                               |
| 1730609637064 | START RequestId: fa33bc2e-d8d1-4f6b-b369-81dc3079d710 Version: $LATEST                                                                                                                                |
| 1730609637077 | END RequestId: fa33bc2e-d8d1-4f6b-b369-81dc3079d710                                                                                                                                                   |
| 1730609637077 | REPORT RequestId: fa33bc2e-d8d1-4f6b-b369-81dc3079d710 Duration: 1048.69 ms Billed Duration: 1049 ms Memory Size: 2048 MB Max Memory Used: 80 MB Status: error Error Type: Runtime.ImportModuleError  |
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

I've tried latest version of CDK and different nodejs versions (see https://github.com/aws-samples/aws-eda-slurm-cluster/issues/279)
I'm at a loss if this is a parallelcluster issue or aws-eda-slurm-cluster issue.  Would like to see if you can reproduce.

I get the same problem if I try to use parallelcluster 3.9.3:
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
|   timestamp   |                                                                                                message                                                                                                |
|---------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 1730626340792 | INIT_START Runtime Version: python:3.12.v36 Runtime Version ARN: arn:aws:lambda:us-west-2::runtime:188d9ca2e2714ff5637bd2bbe06ceb81ec3bc408a0f277dab104c14cd814b081                                   |
| 1730626341838 | [ERROR] Runtime.ImportModuleError: Unable to import module 'CreateParallelClusterConfig': No module named 'rpds.rpds' Traceback (most recent call last):                                              |
| 1730626341904 | INIT_REPORT Init Duration: 1112.04 ms Phase: init Status: error Error Type: Runtime.ImportModuleError                                                                                                 |
| 1730626342851 | [ERROR] Runtime.ImportModuleError: Unable to import module 'CreateParallelClusterConfig': No module named 'rpds.rpds' Traceback (most recent call last):                                              |
| 1730626342920 | INIT_REPORT Init Duration: 1002.02 ms Phase: invoke Status: error Error Type: Runtime.ImportModuleError                                                                                               |
| 1730626342920 | START RequestId: 37d926c6-aea3-4025-b79c-bbf595e460a2 Version: $LATEST                                                                                                                                |
| 1730626342934 | END RequestId: 37d926c6-aea3-4025-b79c-bbf595e460a2                                                                                                                                                   |
| 1730626342934 | REPORT RequestId: 37d926c6-aea3-4025-b79c-bbf595e460a2 Duration: 1014.86 ms Billed Duration: 1015 ms Memory Size: 2048 MB Max Memory Used: 80 MB Status: error Error Type: Runtime.ImportModuleError  |
| 1730626411011 | [ERROR] Runtime.ImportModuleError: Unable to import module 'CreateParallelClusterConfig': No module named 'rpds.rpds' Traceback (most recent call last):                                              |
| 1730626411083 | INIT_REPORT Init Duration: 1009.77 ms Phase: invoke Status: error Error Type: Runtime.ImportModuleError                                                                                               |
| 1730626411083 | START RequestId: 37d926c6-aea3-4025-b79c-bbf595e460a2 Version: $LATEST                                                                                                                                |
| 1730626411096 | END RequestId: 37d926c6-aea3-4025-b79c-bbf595e460a2                                                                                                                                                   |
| 1730626411096 | REPORT RequestId: 37d926c6-aea3-4025-b79c-bbf595e460a2 Duration: 1022.26 ms Billed Duration: 1023 ms Memory Size: 2048 MB Max Memory Used: 80 MB Status: error Error Type: Runtime.ImportModuleError  |
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

I'm not sure I understand how Python3.12.v36 comes into play here. Not sure where that is coming from. Is that somehow involved?  If I look back to May time frame, my successful deployment of 3.9.1 was with 3.9v56.
As noted, I was able to deploy a 3.11.1, but that has slurmctld crashing every 24 hours or so as per filed issue: https://github.com/aws/aws-parallelcluster/issues/6529
That deployment successfully used 3.12.v36.  

3.9.3 deployment is using Rocky 8.9 due to parallelcluster build-image not supporting Rocky8.10 in 3.9.3 due to lack of fsx support which didn't become available until 3.10 for Rocky 8.10.

--- a/source/cdk/cdk_slurm_stack.py
+++ b/source/cdk/cdk_slurm_stack.py
@@ -53,7 +53,7 @@ import base64
 import boto3
 from botocore.exceptions import ClientError
 import config_schema
-from config_schema import get_PARALLEL_CLUSTER_MUNGE_VERSION, get_PARALLEL_CLUSTER_PYTHON_VERSION, get_PC_SLURM_VERSION, get_SLURM_VERSION
+from config_schema import get_PARALLEL_CLUSTER_LAMBDA_RUNTIME, get_PARALLEL_CLUSTER_MUNGE_VERSION, get_PARALLEL_CLUSTER_PYTHON_VERSION, get_PC_SLURM_VERSION, get_SLURM_VERSION
 from constructs import Construct
 from copy import copy, deepcopy
 from hashlib import sha512
@@ -1427,7 +1427,7 @@ class CdkSlurmStack(Stack):
                 aws_lambda.Architecture.X86_64,
             ],
             compatible_runtimes = [
-                aws_lambda.Runtime.PYTHON_3_12,
+                get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parse_version(self.config['slurm']['ParallelClusterConfig']['Version']))
             ],
         )
 
@@ -1437,7 +1437,7 @@ class CdkSlurmStack(Stack):
             function_name=f"{self.stack_name}-CreateBuildFiles",
             description="Create ParallelCluster build configuration files",
             memory_size=2048,
-            runtime=aws_lambda.Runtime.PYTHON_3_12,
+            runtime=get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parse_version(self.config['slurm']['ParallelClusterConfig']['Version'])),
             architecture=aws_lambda.Architecture.X86_64,
             timeout=Duration.minutes(2),
             log_retention=logs.RetentionDays.INFINITE,
@@ -1499,7 +1499,7 @@ class CdkSlurmStack(Stack):
             function_name=f"{self.stack_name}-CreateParallelClusterConfig",
             description="Create ParallelCluster config",
             memory_size=2048,
-            runtime=aws_lambda.Runtime.PYTHON_3_12,
+            runtime=get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parse_version(self.config['slurm']['ParallelClusterConfig']['Version'])),
             architecture=aws_lambda.Architecture.X86_64,
             timeout=Duration.minutes(15),
             log_retention=logs.RetentionDays.INFINITE,
@@ -1547,7 +1547,7 @@ class CdkSlurmStack(Stack):
             function_name=f"{self.stack_name}-CreateParallelCluster",
             description="Create ParallelCluster",
             memory_size=2048,
-            runtime=aws_lambda.Runtime.PYTHON_3_12,
+            runtime=get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parse_version(self.config['slurm']['ParallelClusterConfig']['Version'])),
             architecture=aws_lambda.Architecture.X86_64,
             timeout=Duration.minutes(15),
             log_retention=logs.RetentionDays.INFINITE,
@@ -1846,7 +1846,7 @@ class CdkSlurmStack(Stack):
             function_name=f"{self.stack_name}-CreateHeadNodeARecord",
             description="Create head node A record",
             memory_size=2048,
-            runtime=aws_lambda.Runtime.PYTHON_3_12,
+            runtime=get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parse_version(self.config['slurm']['ParallelClusterConfig']['Version'])),
             architecture=aws_lambda.Architecture.X86_64,
             timeout=Duration.minutes(15),
             log_retention=logs.RetentionDays.INFINITE,
@@ -1893,7 +1893,7 @@ class CdkSlurmStack(Stack):
             function_name=f"{self.stack_name}-UpdateHeadNode",
             description="Update head node",
             memory_size=2048,
-            runtime=aws_lambda.Runtime.PYTHON_3_12,
+            runtime=get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parse_version(self.config['slurm']['ParallelClusterConfig']['Version'])),
             architecture=aws_lambda.Architecture.X86_64,
             timeout=Duration.minutes(15),
             log_retention=logs.RetentionDays.INFINITE,
@@ -1935,7 +1935,7 @@ class CdkSlurmStack(Stack):
                 function_name=f"{self.stack_name}-ConfigUsersGroupsJson",
                 description="Configure users and groups json file",
                 memory_size=2048,
-                runtime=aws_lambda.Runtime.PYTHON_3_12,
+                runtime=get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parse_version(self.config['slurm']['ParallelClusterConfig']['Version'])),
                 architecture=aws_lambda.Architecture.X86_64,
                 timeout=Duration.minutes(15),
                 log_retention=logs.RetentionDays.INFINITE,
@@ -1983,7 +1983,7 @@ class CdkSlurmStack(Stack):
                 function_name=f"{self.stack_name}-ConfigExternalLoginNodes",
                 description="Configure external login nodes",
                 memory_size=2048,
-                runtime=aws_lambda.Runtime.PYTHON_3_12,
+                runtime=get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parse_version(self.config['slurm']['ParallelClusterConfig']['Version'])),
                 architecture=aws_lambda.Architecture.X86_64,
                 timeout=Duration.minutes(15),
                 log_retention=logs.RetentionDays.INFINITE,
@@ -2030,7 +2030,7 @@ class CdkSlurmStack(Stack):
                 function_name=f"{self.stack_name}-DeconfigUsersGroupsJson",
                 description="Deconfigure RES users and groups json file",
                 memory_size=2048,
-                runtime=aws_lambda.Runtime.PYTHON_3_12,
+                runtime=get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parse_version(self.config['slurm']['ParallelClusterConfig']['Version'])),
                 architecture=aws_lambda.Architecture.X86_64,
                 timeout=Duration.minutes(15),
                 log_retention=logs.RetentionDays.INFINITE,
@@ -2072,7 +2072,7 @@ class CdkSlurmStack(Stack):
                 function_name=f"{self.stack_name}-DeconfigExternalLoginNodes",
                 description="Deconfigure external login nodes",
                 memory_size=2048,
-                runtime=aws_lambda.Runtime.PYTHON_3_12,
+                runtime=get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parse_version(self.config['slurm']['ParallelClusterConfig']['Version'])),
                 architecture=aws_lambda.Architecture.X86_64,
                 timeout=Duration.minutes(15),
                 log_retention=logs.RetentionDays.INFINITE,
@@ -2114,7 +2114,7 @@ class CdkSlurmStack(Stack):
             function_name=f"{self.stack_name}-CallSlurmRestApiLambda",
             description="Example showing how to call Slurm REST API",
             memory_size=128,
-            runtime=aws_lambda.Runtime.PYTHON_3_12,
+            runtime=get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parse_version(self.config['slurm']['ParallelClusterConfig']['Version'])),
             architecture=aws_lambda.Architecture.ARM_64,
             timeout=Duration.minutes(1),
             log_retention=logs.RetentionDays.INFINITE,

--- a/source/cdk/config_schema.py
+++ b/source/cdk/config_schema.py
@@ -16,6 +16,9 @@ OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 """
 
+from aws_cdk import (
+    aws_lambda as aws_lambda
+)
 import boto3
 from botocore.client import ClientError
 import json
@@ -317,6 +320,14 @@ def PARALLEL_CLUSTER_SUPPORTS_CUSTOM_MUNGE_KEY(parallel_cluster_version):
 PARALLEL_CLUSTER_SUPPORTS_HOME_MOUNT_VERSION = parse_version('3.8.0')
 def PARALLEL_CLUSTER_SUPPORTS_HOME_MOUNT(parallel_cluster_version):
     return parallel_cluster_version >= PARALLEL_CLUSTER_SUPPORTS_HOME_MOUNT_VERSION
+
+# Version 3.11.0
+
+def get_PARALLEL_CLUSTER_LAMBDA_RUNTIME(parallel_cluster_version):
+    if parallel_cluster_version < parse_version('3.11.1'):
+        return aws_lambda.Runtime.PYTHON_3_9
+    else:
+        return aws_lambda.Runtime.PYTHON_3_12
 
 # Determine all AWS regions available on the account.
 default_region = environ.get("AWS_DEFAULT_REGION", "us-east-1")

